### ISSUE TITLE
Keys do not work on open email window

### ISSUE THREAD

**Issue by Izwbusr** [wshanks/tbkeys#166]
Hi:

Am I missing somethig or does tbkeys only work in the main Thunderbird window?  I am trying to have a key automatically close an open email window.

Thanks

**Comment by Izwbusr** [comment]
Does anybody know if tbkeys should work with open email windows?

thanks

**Comment by morat523035** [comment]
Did you read the README page?

> There are separate fields in the preferences page for setting key bindings for the main Thunderbird window and the compose window. Key bindings do not fire in other windows.

README - Customizing key bindings
http://github.com/wshanks/tbkeys#customizing-key-bindings

The main window and compose window are supported. The message window is not supported.

How to add key bindings support for message window
http://forums.mozillazine.org/viewtopic.php?p=14919152#p14919152

I don't know the hack for tbkeys 2.2.4. (latest release)

**Comment by wshanks** [comment]
I don't think it would be too hard to support the message window. I had never considered it before -- I just open messages in tabs. I think it just takes adding the id of the message window to this object:

https://github.com/wshanks/tbkeys/blob/568a5d494aa8ab8c989c343e59b11ec7280cd2ff/addon/implementation.js#L33-L38

I think the message window could be treated like the "main" window and not as a third set of bindings?

**Comment by morat523035** [comment]
If you want to share bindings with the main window, then I can always create commands like so.

```
(function () {
  if (window.location.href == 'chrome://messenger/content/messenger.xhtml') {
    window.CloseTabOrWindow();
  } else if (window.location.href == 'chrome://messenger/content/messageWindow.xhtml') {
    window.close();
  }
})();
```

Since not all my commands work in both windows.

Similar issue
http://github.com/wshanks/tbkeys/issues/77

**Comment by wshanks** [comment]
Fair enough @morat523035! I forgot this had been brought up before three years ago :grimacing:
